src/coreclr/jit/optimizer.cpp

```diff
@@ -6861,9 +6862,7 @@ bool Compiler::optHoistThisLoop(unsigned lnum, LoopHoistContext* hoistCtxt, Basi
     }
 #endif
 
-    unsigned floatVarsCount = VarSetOps::Count(this, lvaFloatVars);
-
-    if (floatVarsCount > 0)
+    if (!VarSetOps::IsEmpty(this, lvaFloatVars))
```

Comment:
It looks like `lvaFloatVars` doesn't account for `SIMD` variables (which end up in the same registers and I'd imagine have similar considerations). Is that expected and/or should they be tracked separately?

Reply:
This PR is simply a refactoring of existing code. I've opened https://github.com/dotnet/runtime/issues/82351 to track this issue. It looks like SIMD type trees get treated as using `int` registers instead of `float` registers for the purpose of loop hoisting profitability.

Reply:
Thanks! Definitely not something I thought should be handled here, more just an observation/question of the code that was being refactored